cli/src/wallet_cli/mod.rs

```diff
@@ -818,11 +821,12 @@ pub async fn sync_command(wallet: &Wallet) -> Result<(), Error> {
 
 /// `transaction` command
 pub async fn transaction_command(wallet: &Wallet, selector: TransactionSelector) -> Result<(), Error> {
-    let mut transactions = wallet.transactions().await;
+    let wallet_data = wallet.data().await;
     let transaction = match selector {
-        TransactionSelector::Id(id) => transactions.into_iter().find(|tx| tx.transaction_id == id),
+        TransactionSelector::Id(id) => wallet_data.transactions().find(|tx| tx.transaction_id == id),
```

Comment:
I think it's a bit weird that `WalletData::outputs` return the hashmap and `WelletData::transactions` returns an iterator